File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/model/body/parking/GetParkingFeeForFleetBody.java
package com.lattis.ellipse.data.network.model.body.parking;

import com.google.gson.annotations.SerializedName;
import com.lattis.ellipse.domain.model.Location;

/**
 * Created by ssd3 on 8/1/17.
 */

public class GetParkingFeeForFleetBody {

    @SerializedName("latitude")
    private double latitude;
    @SerializedName("longitude")
    private double longitude;
    @SerializedName("accuracy")
    private double accuracy;
    @SerializedName("fleet_id")
    private int fleet_id;

    public GetParkingFeeForFleetBody(Location location,
                           int fleet_id) {
        this.latitude = location.getLatitude();
        this.longitude = location.getLongitude();
        if(location.hasAccuracy()) {
            this.accuracy = location.getAccuracy();
        }else{
            this.accuracy = 0.0;
        }
        this.fleet_id = fleet_id;
    }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/entity/body/parking/GetParkingFeeForFleetBody.kt
package com.lattis.data.entity.body.parking

import com.google.gson.annotations.SerializedName
import com.lattis.domain.models.Location

class GetParkingFeeForFleetBody(
    location: Location,
    fleet_id: Int
) {
    @SerializedName("latitude")
    private val latitude: Double

    @SerializedName("longitude")
    private val longitude: Double

    @SerializedName("accuracy")
    private var accuracy = 0.0

    @SerializedName("fleet_id")
    private val fleet_id: Int

    init {
        latitude = location.latitude
        longitude = location.longitude
        if (location.hasAccuracy()) {
            accuracy = location.accuracy.toDouble()
        } else {
            accuracy = 0.0
        }
        this.fleet_id = fleet_id
    }
}